starknet_py/utils/typed_data.py

```diff
@@ -28,9 +107,17 @@ class TypedData:
 
     types: Dict[str, List[Parameter]]
     primary_type: str
-    domain: StarkNetDomain
+    domain: Domain
     message: dict
 
+    def __post_init__(self):
+        self._verify_types()
+
+    def _hash_method(self) -> HashMethod:
+        if self.domain.resolved_revision is Revision.V0:
+            return HashMethod.PEDERSEN
+        return HashMethod.POSEIDON
+
```

Comment:
I'm not a fan that we each time when we want to retrieve hash method, we need to call `_hash_method()`. However, having `hash_method` as a field, would require from use to remove `frozen=True` from `TypedData`. Considering this fact, imo current solution is better.

Reply:
I think that's fine. We can also add a `@property` decorator and then it can be accessed without parentheses.